src/core/expr/fexpr_sumprod.cc

```diff
@@ -0,0 +1,131 @@
+//------------------------------------------------------------------------------
+// Copyright 2022 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+#include "column/const.h"
+#include "column/sumprod.h"
+#include "documentation.h"
+#include "expr/fexpr_func.h"
+#include "expr/eval_context.h"
+#include "expr/workframe.h"
+#include "python/xargs.h"
+#include "stype.h"
+
+namespace dt {
+namespace expr {
+
+
+template <bool SUM>
+class FExpr_SumProd : public FExpr_Func {
```

Comment:
Since `sum()` and `prod()` are both unary functions, I guess they should be better inherited from `FExpr_FuncUnary`.

Reply:
We use `FExpr_Func` for cumsum and cumprod, that's why I used it here. So single functions should inherit from func_unary? Any suggestions on usage?

Reply:
Right. It seems that `FExpr_FuncUnary` is currently not useful for reducers or cumulative functions, because it doesn't take into account the groupby information. We probably need to implement `FExpr_FuncUnaryReducer` (or something similar (not as a part of this PR).